extensions/resource-deployment/src/services/tools/toolBase.ts

```diff
@@ -121,10 +121,10 @@ export abstract class ToolBase implements ITool {
 		return this.status !== ToolStatus.Installed;
 	}
 
-	public get storagePath(): string {
+	public async getStoragePath(): Promise<string> {
 		const storagePath = this._platformService.storagePath();
 		if (!this._storagePathEnsured) {
-			this._platformService.ensureDirectoryExists(storagePath);
+			await this._platformService.ensureDirectoryExists(storagePath);
```

Comment:
I'd recommend to move the ensureDirectoryExists to the resourceTypePickerDialog so that we don't have to this all over the places

Reply:
In that case, the clean way to do that would for the tools to start accepting storagePath as a constructor parameter. Do you prefer that approach?

Reply:
not sure how will that help, accepting it as constructor parameter doesn't mean it already exists, right?

Reply:
You can still have the platform service store it - but the platform service should be the one ensuring it exists. Right now if platformService.storagePath was to change for some reason this code would no longer be guaranteeing that the directory exists which is a separate issue. 

So it'd be like -> initialize storage path at startup - > guarantee it exists -> then initialize tools

---
In reply to: [343905474](https://github.com/microsoft/azuredatastudio/pull/8250#discussion_r343905474) [](ancestors = 343905474)

Reply:
I absolutely agree!